43 create editing profile

Bivis63 opened the pull request.

Bivis63 committed:
Добавил экран редактирования личных данных
1.Нету кастомной кнопки для "Удалить аккаунт"
2.Нету касмотной кнопки для "Отменить" в bottomSheet
3.BottomSheet добавил через навигацию, можно ли оставить диалоги как есть или тоже нужно добавить через навигацию?

Bivis63 committed:
Добавил viewModel и зависимости

n-shvedov27 commented on app/src/main/java/com/innoprog/android/feature/editingprofile/presentation/EditingProfileFragment.kt:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    private fun showDialogForExit() {
+        val dialog = Dialog(requireContext())
```
А эти диалоги почему не через навигацию?

n-shvedov27 commented on app/src/main/java/com/innoprog/android/feature/editingprofile/presentation/BottomSheetFragment.kt:
```diff
@@ -0,0 +1,54 @@
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.innoprog.android.databinding.FragmentBottomSheetBinding
+
+class BottomSheetFragment : BottomSheetDialogFragment() {
```
Давай назовем его абстрактным BaseBottomSheetFragment 
C возможностью настраивать нужный layout
По сути как в BaseFragment
Жаль его в целом от BaseFragment наследовать не получится

n-shvedov27 replied:
Ну и вынести его в папку base
А сюда добавить реализацию

n-shvedov27 commented on app/src/main/res/layout/fragment_editing_profile.xml:
```diff
@@ -0,0 +1,211 @@
+            android:layout_marginTop="@dimen/margin_16"
+            android:layout_marginBottom="@dimen/margin_30"
+            android:gravity="center"
+            app:buttonType="def"
```
Ты же вроде хотел добавить настройку цвета текста кнопки

n-shvedov27 commented on app/src/main/res/navigation/auth_nav_graph.xml:
```diff
@@ -3,7 +3,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/nav_graph"
-    app:startDestination="@id/authorizationFragment">
+    app:startDestination="@id/editingProfileFragment">
```
Я не смогу эту в dev слить
editingProfileFragment должен открываться через ProfileFragment

n-shvedov27 commented on app/src/main/java/com/innoprog/android/feature/editingprofile/presentation/EditingProfileFragment.kt:
```diff
@@ -0,0 +1,94 @@
+package com.innoprog.android.feature.editingprofile.presentation
```
Я бы лучше так пекеджи расположил
<img width="364" alt="Снимок экрана 2024-03-26 в 10 53 41" src="https://github.com/InnoProg-Project/InnoProg/assets/37715977/c4422d99-7007-4fc5-9027-1ba6f5e7a9b6">

n-shvedov27 commented on uikit/src/main/res/values/strings.xml:
```diff
@@ -4,4 +4,19 @@
+    <string name="delete_account">Удалить аккаунт?</string>
+    <string name="cancellation">Отмена</string>
+    <string name="go_out">Выйти</string>
+    <string name="delete">Удалить</string>
```
Не углядел
Строки нужные для конкретной фичи не должны быть в uikit'е

Добавь плз в app файл
editprofile_string.xml

И туда складывай нужные строки

Bivis63 committed:
Добавил в кнопку удалить в кастом вью для кнопок
сделал диалоги через навигацию
создал BaseFragment для bottomsheet и диалоговых окон
вынес названия в editprofile.string

Bivis63 committed:
вернул стартовый фрагмент на авторизацию

n-shvedov27 reviewed (changes requested):
Конфликты

Bivis63 committed:
Merge remote-tracking branch 'origin/develop' into 43-create_editing_profile

# Conflicts:
#	app/src/main/res/values/strings.xml
#	uikit/src/main/res/values/dimens.xml

n-shvedov27 commented on uikit/src/main/java/com/innoprog/android/uikit/InnoProgButtonView.kt:
```diff
@@ -100,6 +100,11 @@ class InnoProgButtonView @JvmOverloads constructor(
                 alpha = if (state == ButtonState.ENABLED) NOT_TRANSPARENT else HALF_TRANSPARENT
             }
+
+            ButtonType.DELETE -> {
```
Давай лучше сделаем проставление цвета кнопки через отдельный метод, а не еще один ButtonType

Bivis63 replied:
добавил

n-shvedov27 commented on app/src/main/res/layout/fragment_editing_profile.xml:
```diff
@@ -0,0 +1,217 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <androidx.appcompat.widget.Toolbar
```
Почему не InnoProgTopBarView?

Bivis63 replied:
мне кажется она не очень похожа на то что на макете )

Bivis63 replied:
поставил кастомный, но очень странно что по макету все размеры и стили сходятся а шрифт выглядит как то иначе

n-shvedov27 replied:
Завел задачку на это
https://github.com/InnoProg-Project/InnoProg/issues/95

Bivis63 committed:
порешал конфликты
сделал метод для смены цвета текста у кнопок
поставил кастомный тул бар

Bivis63 committed:
порешал конфликты
сделал метод для смены цвета текста у кнопок
поставил кастомный тул бар
